Dockerfile

```diff
@@ -20,13 +27,30 @@ RUN ln -sf /dev/stdout /var/log/nginx/access.log && \
 # Create directories
 RUN mkdir -p /var/www/app /usr/share/nginx_auth /usr/share/nginx_crt /opt/share/webapp
 
-#copy code and install node & bower deps
+# Trying to exploit docker container layers:
+# If the package.json file changes then Docker will re-run the npm install sequence...
+# otherwise our modules are cached so we aren't rebuilding them every time we change our apps source code!
+COPY package.json /tmp/package.json
+# there are some dependencies not in npm library yet
```

Comment:
on second thought, it may happen again that you have something in components and not on npm, right?
probably safer to keep it that way.